js/DataService.js

```diff
@@ -8,13 +8,46 @@ angular.module('PathOfDamage')
     var int = parseInt(string);
     return isNaN(int) ? null : int;
   }
+  
+  function Row(key, value) {
+    this.enabled = true;
+    this.name = '';
+    this.value = null;
+    if (key !== undefined) {
```

Comment:
I'm not a huge fan of this, but it makes the rest of the code very concise and clean